src/fallback_language.js

```diff
@@ -0,0 +1,12 @@
+angular.module("gettext").factory("gettextFallbackLanguage", function () {
+
+    var pattern = /([^_]+)_[^_]+$/;
+
+    return function (langCode) {
+        var matches = pattern.exec(langCode);
+        if (matches){
+            return matches[1];
+        }
+        return null;
+    };
+});
```

Comment:
This might be executed *a lot*. I suggest adding a simple memoization:

```js
angular.module("gettext").factory("gettextFallbackLanguage", function () {
    var cache = {};
    var pattern = /([^_]+)_[^_]+$/;

    return function (langCode) {
        if (cache[langCode]) {
            return cache[langCode];
        }

        var matches = pattern.exec(langCode);
        if (matches){
            cache[langCode] = matches[1];
            return matches[1];
        }

        return null;
    };
});
```